pgjdbc/src/main/java/org/postgresql/util/internal/LeakTracker.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util.internal;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.function.Consumer;
+
+public class LeakTracker<T> {
```

Comment:
any reason not to make this final? It can always be opened up if we need to extend in future.

Reply:
It is "internal", so is it really that important?